cli/src/cmd/cast/create2.rs

```diff
@@ -0,0 +1,150 @@
+//! cast create2 subcommand
+
+use crate::cmd::Cmd;
+use cast::SimpleCast;
+use clap::Parser;
+use ethers::{
+    core::rand::thread_rng,
+    types::{Address, Bytes, H256, U256},
+    utils::get_create2_address_from_hash,
+};
+use eyre::{Result, WrapErr};
+use rayon::prelude::*;
+use regex::RegexSetBuilder;
+use std::time::Instant;
+
+#[derive(Debug, Clone, Parser)]
+pub struct Create2Args {
+    #[clap(
+        long,
+        short,
+        help = "Prefix for the contract address.",
+        required_unless_present_any = &["ends-with", "matching"],
+        value_name = "HEX"
+    )]
+    starts_with: Option<String>,
+    #[clap(long, short, help = "Suffix for the contract address.", value_name = "HEX")]
+    ends_with: Option<String>,
+    #[clap(long, short, help = "Sequence that the address has to match", value_name = "HEX")]
+    matching: Option<String>,
+    #[clap(short, long)]
+    case_sensitive: bool,
+    #[clap(
+        short,
+        long,
+        help = "Address of the contract deployer.",
+        default_value = "0x4e59b44847b379578588920ca78fbf26c0b4956c",
+        value_name = "ADDRESS"
+    )]
+    deployer: Address,
+    #[clap(
+        short,
+        long,
+        help = "Init code of the contract to be deployed.",
+        value_name = "HEX",
+        default_value = ""
+    )]
+    init_code: String,
+}
+
+impl Cmd for Create2Args {
+    type Output = ();
+
+    fn run(self) -> eyre::Result<Self::Output> {
+        let Create2Args { starts_with, ends_with, matching, case_sensitive, deployer, init_code } =
+            self;
+        Create2Args::generate_address(
+            starts_with,
+            ends_with,
+            matching,
+            case_sensitive,
+            deployer,
+            init_code,
+        )
+    }
```

Comment:
it doesn't make much sense to split all fields and then provide all of them as args for `Create2Args::generate_address`

suggesting to make `Create2Args::generate_address(self)` instead and move the split `let Create2Args { starts_with, ends_with, matching, case_sensitive, deployer, init_code } =
            self;` there